nomad/structs/actions.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package structs
+
+import "slices"
+
+type Action struct {
```

Comment:
Hum...I feel like this name may be too generic 🤔 

The `structs` package deal with all sorts of stuff and a struct called `Action` may not be obvious what it relates to. Perhaps `TaskAction` would be a better name?

Reply:
I'm not opposed to this (they started out with that name!) but lately we've been talking about Node Actions and similar concepts, which I think would borrow from this a bit. I'm torn!

Reply:
Are they similar or the _same_? 

Reusing structs for similar concepts may result in weird interfaces, like attributes that are `nil` in some cases but not in others. And sometimes that's OK. So there's not right answer, just _feels_ at this point. 

My vote would be for `TaskAction`, but I will leave it up to you 😄